src/main/asciidoc/java/http.adoc

```diff
@@ -1150,6 +1389,77 @@ httpServer.requestHandler(request -> {
 });
 ----
 
+==== Client push
+
+Server push is a new feature of HTTP/2 that enables sending multiple responses in parallel for a single client request.
+
+A push handler can be set on a request to receive the request/response pushed by the server:
+
+[source,java]
+----
+HttpClientRequest request = client.get("/index.html", response -> {
+  // Process index.html response
+});
+
+// Set a push handler to be aware of any resource pushed by the server
+request.pushHandler(pushedRequest -> {
+
+  // A resource is pushed for this request
+  System.out.println("Server pushed " + pushedRequest.path());
+
+  // Set an handler for the response
+  pushedRequest.handler(pushedResponse -> {
+    System.out.println("The response for the pushed request");
+  });
+});
+
+// End the request
+request.end();
+----
+
+If the client does not want to receive a pushed request, it can reset the stream:
+
+[source,java]
+----
+HttpClientRequest request = client.get("/index.html", response -> {
```

Comment:
yes wrong one :-)

> On Apr 6, 2016, at 10:06 AM, Clement Escoffier <notifications@github.com> wrote:
> 
> In src/main/asciidoc/java/http.adoc <https://github.com/eclipse/vert.x/pull/1353#discussion_r58665850>:
> 
> > +
> > +  // Set an handler for the response
> > +  pushedRequest.handler(pushedResponse -> {
> > +    System.out.println("The response for the pushed request");
> > +  });
> > +});
> > +
> > +// End the request
> > +request.end();
> > +----
> > +
> > +If the client does not want to receive a pushed request, it can reset the stream:
> > +
> > +[source,java]
> > +----
> > +HttpClientRequest request = client.get("/index.html", response -> {
> Wrong snippet ?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/eclipse/vert.x/pull/1353/files/c7a6a1ce8a1da75fc0edcccf686f25b82162b149#r58665850>